transport/tchannel/config.go

```diff
@@ -92,6 +94,10 @@ func (ts *transportSpec) Spec() config.TransportSpec {
 
 func (ts *transportSpec) buildTransport(tc *TransportConfig, k *config.Kit) (transport.Transport, error) {
 	var cfg transportConfig
+	// Default configuration.
+	cfg.tracer = opentracing.GlobalTracer()
+	cfg.name = k.ServiceName()
```

Comment:
Don't change the name. Check the rest of the function. We expect the name to be unset on line 105 and we explicitly set it on line 117. The idea is that users cannot change the service name using TChannel options.